### DISCUSSION TITLE
[Run issue]: Makefile:723: recipe for target 'cmake_check_build_system' failed make: *** [cmake_check_build_system] Error 127

### DISCUSSION THREAD

**Discussion by ghost** [OpenTOPAS/OpenTOPAS#89] [Run-Problem] [answered]
### Contact Details

cecilianovawijaya01@gmail.com

### What was the issue?

Hello,

I was trying to install an extension, but I’m not sure why it is not working. I’ve installed other extensions (like TOPAS-nBio) before without any issues, so I’m unsure why it’s not working this time.

Here’s the error message I received:
```
(base) mirlab@mirlab-TRX40-AORUS-MASTER:~/Applications/TOPAS/OpenTOPAS-build$ ccmake ../OpenTOPAS -DCMAKE_INSTALL_PREFIX=../OpenTOPAS-install

(base) mirlab@mirlab-TRX40-AORUS-MASTER:~/Applications/TOPAS/OpenTOPAS-build$ sudo make -j20 install
make: /snap/cmake/1403/bin/cmake: Command not found
Makefile:723: recipe for target 'cmake_check_build_system' failed
make: *** [cmake_check_build_system] Error 127
```

My CMake version is `cmake version 3.30.5`.

I’ve tried uninstalling and reinstalling CMake, but that didn’t resolve the issue.

Can anyone help me figure out how to fix this?

Thank you for your attention.

### TOPAS category

Geometry

### Version

4.0

### What Operative System (OS) are you running on?

Linux

### Relevant log output

```shell
(base) mirlab@mirlab-TRX40-AORUS-MASTER:~/Applications/TOPAS/OpenTOPAS-build$ ccmake ../OpenTOPAS -DCMAKE_INSTALL_PREFIX=../OpenTOPAS-install

(base) mirlab@mirlab-TRX40-AORUS-MASTER:~/Applications/TOPAS/OpenTOPAS-build$ sudo make -j20 install
make: /snap/cmake/1403/bin/cmake: Command not found
Makefile:723: recipe for target 'cmake_check_build_system' failed
make: *** [cmake_check_build_system] Error 127
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by icmeyer** [comment]
Sorry installing the extension is giving you some trouble. Looking at your post a few things stand out as potential problems. Also we might need more info to debug.

1. I am not familiar with `ccmake` but using just `cmake` should also produce relevant output on the configuration that might help you debug. If you could post it here too that would be helpful.
2. You should not have to use `sudo` with your `make install` if you have the proper permissions for your install path. Using `sudo` can mess with your path configuration, this is my best guess at the problem: `ccmake` without sudo specifies a cmake executable not in your sudo path. So you either have to `sudo cmake && sudo make install` or `cmake && make install` and I highly recommend removing `sudo`
3. Which extension are you installing? Is it a publicly available one or one you are building yourself? Doesn't seem to be a problem with the extension itself so far, but if you were able to install nBio with the same procedure it is a bit suspect

Good luck debugging!

**Reply by ghost** [reply]
Hello!

Thank you for your response.

I was using `ccmake` just to open the configuration setup. It’s the same as `cmake`, but it allows me to open the configuration setup like this:
![image](https://github.com/user-attachments/assets/e6571721-e282-43ac-ba76-fb49d88009ea)
I tried your suggestion to remove `sudo`, but the same issue still occurs:
![image](https://github.com/user-attachments/assets/48178fe2-036a-4876-8e9f-87bc6fc6bb9b)
I am installing the extensions from this [GitHub repository](https://github.com/AKlapproth/MultiScale_AuNP_TOPAS.git). I have successfully installed it on my own computer, but now I’m trying to use the computer in my lab and it suddenly doesn’t work. I’m not sure why.

Thank you for your help! 

Please let me know if you have any suggestions on how to fix this.

**Comment by leehoy** [comment]
Hello. Did you copy the original OpenTOPAS-{source, build} to the new system? If so, the existing CMake configurations from the original system may cause errors due to discrepancies between the systems. Can you run cmake and make after removing following files and directories: "CMakeCache.txt", "CMakeFiles", "cmake_install.cmake", and "Makefile"?

**Reply by ghost** [reply]
Hello!

Thank you for your answer.

I didn’t copy the original OpenTOPAS-{source, build} to the new system. Instead, I reinstalled everything following the [installation guide](https://github.com/OpenTOPAS/OpenTOPAS/blob/main/OpenTOPAS_quickStart_Debian.md). Previously, it worked well, and I could easily install new extensions without any issues. Suddenly, it doesn't work this time.

I tried your suggestion of removing the files, but the error persisted. Here’s the terminal output:
![Selection_051](https://github.com/user-attachments/assets/c1031ede-e891-41b2-9e73-a95eea951292)

However, your suggestion gave me an idea, and I ended up completely removing OpenTOPAS and reinstalling it. It fixed the issue, but I’m still not entirely sure why the error occurred or how the reinstallation resolved it. I don’t know if this means that if the issue happens again in the future, we will always need to remove the program and reinstall it each time.

**Reply by leehoy** [reply, marked as answer]
I'm glad to hear that you have resolved the issue. 
I just noticed that "Build files have been written to: /home.../TOPAS/OpenTOPAS" at the end of your terminal output. But it should be written to ".../TOPAS/OpenTOPAS-build". 
I guess this has something to do with the issues.
